pkg/controller/buildrun/buildrun_controller.go

```diff
@@ -276,70 +284,72 @@ func (r *ReconcileBuildRun) retrieveServiceAccount(ctx context.Context, build *b
-		op, err := controllerutil.CreateOrUpdate(context.TODO(), r.client, serviceAccount, func() error {
+		ctxlog.Info(ctx, "create or update serviceAccount for BuildRun", namespace, buildRun.Namespace, name, serviceAccountName, "BuildRun", buildRun.Name)
+		op, err := controllerutil.CreateOrUpdate(ctx, r.client, serviceAccount, func() error {
 			serviceAccount.SetLabels(map[string]string{buildv1alpha1.LabelBuildRun: buildRun.Name})
 
 			ownerReference := metav1.NewControllerRef(buildRun, buildv1alpha1.SchemeGroupVersion.WithKind("BuildRun"))
 			serviceAccount.SetOwnerReferences([]metav1.OwnerReference{*ownerReference})
 
-			ApplyCredentials(build, serviceAccount)
+			ApplyCredentials(ctx, build, serviceAccount)
 
 			return nil
 		})
 		if err != nil {
 			return nil, err
 		}
-		ctxlog.Info(ctx, "Automatic generation of service account", "ServiceAccount", serviceAccount.Name, "Operation", op)
+		ctxlog.Debug(ctx, "Automatic generation of service account", namespace, serviceAccount.Namespace, name, serviceAccount.Name, "Operation", op)
 	} else {
 		// If ServiceAccount or the name of ServiceAccount in buildRun is nil, use pipeline serviceaccount
 		if buildRun.Spec.ServiceAccount == nil || buildRun.Spec.ServiceAccount.Name == nil {
 			serviceAccountName := pipelineServiceAccountName
-			err := r.client.Get(context.TODO(), types.NamespacedName{Name: serviceAccountName, Namespace: buildRun.Namespace}, serviceAccount)
+			err := r.client.Get(ctx, types.NamespacedName{Name: serviceAccountName, Namespace: buildRun.Namespace}, serviceAccount)
 			if err != nil && !apierrors.IsNotFound(err) {
 				return nil, err
 			} else if apierrors.IsNotFound(err) {
 				serviceAccountName = defaultServiceAccountName
-				err = r.client.Get(context.TODO(), types.NamespacedName{Name: serviceAccountName, Namespace: buildRun.Namespace}, serviceAccount)
+				ctxlog.Info(ctx, "falling back to default serviceAccount", namespace, buildRun.Namespace)
+				err = r.client.Get(ctx, types.NamespacedName{Name: serviceAccountName, Namespace: buildRun.Namespace}, serviceAccount)
 				if err != nil {
 					return nil, err
 				}
 			}
 		} else {
 			serviceAccountName := *buildRun.Spec.ServiceAccount.Name
-			err := r.client.Get(context.TODO(), types.NamespacedName{Name: serviceAccountName, Namespace: buildRun.Namespace}, serviceAccount)
+			err := r.client.Get(ctx, types.NamespacedName{Name: serviceAccountName, Namespace: buildRun.Namespace}, serviceAccount)
 			if err != nil {
 				return nil, err
 			}
 		}
 
 		// Add credentials and update the service account
-		if modified := ApplyCredentials(build, serviceAccount); modified {
-			if err := r.client.Update(context.TODO(), serviceAccount); err != nil {
+		if modified := ApplyCredentials(ctx, build, serviceAccount); modified {
+			ctxlog.Info(ctx, "update ServiceAccount with secrets from build", namespace, serviceAccount.Namespace, name, serviceAccount.Name)
```

Comment:
I would like to confirm when using `xxx + ing` and when using `xxx`?

Follow the logging format above, I think this one should be `updating ServiceAccount ...`

Reply:
both are fine for me